### PR TITLE
pre_defined_classes-simple_programs_taking_numeric_user_input-hard-q1

### PR DESCRIPTION
**Pull request by Hedgemon4**
…to finish

### CODE REVIEW

File: src/pre_defined_classes/simple_programs_taking_user_input/hard/q1/PayrollTaxTest.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import global.variables.clauses.NewLine;
+import global.variables.clauses.StringLiteral;
+import org.junit.jupiter.api.RepeatedTest;
+
+import java.nio.channels.NonWritableChannelException;
+
+public class PayrollTaxTest extends BaseTest {
+
+    public Clause[] testSentence() {
+        provideInput("48.75 59.86 0.16 ");
```
Comment (SanaBob) [inline src/pre_defined_classes/simple_programs_taking_user_input/hard/q1/PayrollTaxTest.java:15]: not sure if it works if you put it here, it should be in the run main method. Also if you're inputting multiple variables it has to be separated by System.lineSeparator()

Comment (Hedgemon4) [reply src/pre_defined_classes/simple_programs_taking_user_input/hard/q1/PayrollTaxTest.java:15]: Actually it is working without that

**Review by SamamthaH — DISMISSED** [review]

File: src/pre_defined_classes/simple_programs_taking_user_input/hard/q1/MainTest.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import global.variables.Clause;
+import global.variables.clauses.DoubleLiteral;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.StringLiteral;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class MainTest extends BaseTest {
+    // Parsons (with distractors)
```
Comment (aadiraju) [inline src/pre_defined_classes/simple_programs_taking_user_input/hard/q1/MainTest.java:13]: remove the brackets to be consistent with other questions

File: src/pre_defined_classes/simple_programs_taking_user_input/hard/q1/MainTest.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+                new NewLine(),
+                new StringLiteral("Net Earnings: "),
+                new DoubleLiteral("net")
+        };
+    }
+
+    public void runMain() {
+        provideInput("48.75 59.86 0.16 ");
+        PayrollTax.main(new String[0]);
+    }
```
Comment (aadiraju) [inline src/pre_defined_classes/simple_programs_taking_user_input/hard/q1/MainTest.java:36]: I think you should rewrite this question with InputTest

**Review by aadiraju — COMMENTED** [review]
The tests seem fine, but some minor changes and you need to rewrite it with Input test

File: src/pre_defined_classes/simple_programs_taking_user_input/hard/q1/MainTest.java (hunk 3 above)
Comment (Hedgemon4) [reply src/pre_defined_classes/simple_programs_taking_user_input/hard/q1/MainTest.java:36]: ok

**Review by SamamthaH — COMMENTED** [review]
Is there a way to get rid of the extra files not pertaining to the unit test? If not, then it looks good to me.

**Review by SamamthaH — APPROVED** [review]

**Review by aadiraju — APPROVED** [review]
Looks good! I'll let @Opeyem1a merge

**Review by Opeyem1a — APPROVED** [review]

**Pull request merged**